multimodal/src/autogluon/multimodal/predictor.py

```diff
@@ -2475,27 +2479,51 @@ def export_onnx(
 
         device_type = "cuda" if torch.cuda.is_available() else "cpu"
         device = torch.device(device_type)
-        self._model.to(device).eval()
-        batch = move_to_device(batch, device=device)
+        input_keys = self._model.input_keys
```

Comment:
This assignment seems unnecessary.

Reply:
This variable is used in the following for loop

```python
for key in input_keys:
    inp = batch[key]
    # support mixed precision on floating point inputs, and leave integer inputs (for language models) untouched.
    if inp.dtype.is_floating_point:
        batch[key] = inp.to(device, dtype=dtype)
    else:
        batch[key] = inp.to(device)
```

Reply:
Yes, line 2481 uses self._model.input_keys.